tests/functionality/testSysRandGMRES.cpp

```
@@ -177,7 +198,7 @@ int test(int argc, char *argv[])
     std::cout << "Result is not a finite number!\n";
     error_sum++;
   }
-  if (final_norm/norm_b > (10.0 * tol)) {
+  if (final_norm/norm_b > (10.0 * tol_out)) {
```

Maybe this `10.0` should be an optional parameter? I can imagine a scenario where I specify a small tolerance, but don't necessarily want to throw an error if I wasn't able to achieve it (or 10X it)

---

See #203, for example. We do need to make these tests more modular and more configurable.